clicommand/pipeline_upload.go

```diff
@@ -176,27 +175,31 @@ var PipelineUploadCommand = cli.Command{
 		ctx, cfg, l, _, done := setupLoggerAndConfig[PipelineUploadConfig](ctx, c)
 		defer done()
 
-		// Find the pipeline either from STDIN or the first argument
-		var input *os.File
-		var filename string
+		// Find the pipeline either from STDIN or the non-flag arguments
+		type input struct {
+			file *os.File
+			name string
+		}
+		var inputs []input
 
 		switch {
-		case cfg.FilePath != "":
-			l.Info("Reading pipeline config from %q", cfg.FilePath)
+		case len(cfg.FilePaths) > 0:
+			l.Info("Reading pipeline configs from %q", cfg.FilePaths)
 
-			filename = filepath.Base(cfg.FilePath)
-			file, err := os.Open(cfg.FilePath)
-			if err != nil {
-				return fmt.Errorf("failed to read file: %w", err)
+			for _, fn := range cfg.FilePaths {
+				file, err := os.Open(fn)
+				if err != nil {
+					return fmt.Errorf("failed to read file: %w", err)
+				}
+				defer file.Close()
```

Comment:
With the `defer` within the loop, should the loop content be wrapped within `func() {` to properly ensure each file handle is closed correctly?

Reply:
Hmmmm will the defer still then keep the reference correctly for each of the individual files that have been opened? For some reason I was thinking that only the last loop would have the correct defer retained.

Reply:
https://go.dev/play/p/tijE6PJW00m